app/models/legal_aid_application.rb

```diff
@@ -54,12 +54,17 @@ def bank_transactions
 
   def self.find_by_secure_id!(secure_id)
     secure_data = SecureData.for(secure_id)
-    find_by! secure_data[:legal_aid_application]
+    if secure_data[:expired_at] && secure_data[:expired_at] < Time.current
+      SimpleResult.new(error: :expired)
+    else
+      SimpleResult.new(value: find_by!(secure_data[:legal_aid_application]))
+    end
   end
 
   def generate_secure_id
     SecureData.create_and_store!(
       legal_aid_application: { id: id },
+      expired_at: Time.current.beginning_of_day + 8.days,
```

Comment:
`8` is a magic number. As this is an important configuration setting it really should be in a constant and declared near the top of this model:
```ruby
SECURE_ID_DAYS_TO_EXPIRE = 8
```
Then here:
```
expired_at: Time.current.beginning_of_day + SECURE_ID_DAYS_TO_EXPIRE.days,
```